### PR TITLE
STYLE: Extending codespell to pandas/tests/  part3  38802

### LINKED ISSUE #38802: CI/STYLE extend codespell beyond pandas/core
There is task left here:

1. apply the following diff
```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index d78c2bacc..bcb9202d4 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -173,4 +173,4 @@ repos:
     hooks:
     -   id: codespell
         types_or: [python, rst, markdown]
         files: ^(pandas|doc)/
-        exclude: ^pandas/tests/
```
, run `pre-commit run codespell --all-files`, and fixup the mistakes 

You _may_ need you to add new words to ignore in `setup.cfg`.

### PR DESCRIPTION
**Pull request by 01-vyom**
- [x] closes #38802
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Legend, thanks! Looks good to me pending green

File: pandas/tests/io/xml/test_to_xml.py [hunk 1]
```diff
@@ -411,12 +411,12 @@ def test_attrs_cols_prefix(datapath, parser):
 
 def test_attrs_unknown_column(parser):
     with pytest.raises(KeyError, match=("no valid column")):
-        geom_df.to_xml(attr_cols=["shape", "degreees", "sides"], parser=parser)
+        geom_df.to_xml(attr_cols=["shape", "degrees", "sides"], parser=parser)
```
Comment (MarcoGorelli) [inline pandas/tests/io/xml/test_to_xml.py:414]: Looks like this is failing a test, might have to choose a different word

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by 01-vyom** [comment]
Done with all the changes.

**Comment by MarcoGorelli** [comment]
Thanks @01-vyom

**Pull request merged**